pkg/config/model/types.go

```diff
@@ -111,7 +111,15 @@ type Setup interface {
 	SetEnvPrefix(in string)
 	BindEnv(input ...string)
 	SetEnvKeyReplacer(r *strings.Replacer)
+
+	// SetEnvKeyTransformer is deprecated in favor of LoadEnv* functions
 	SetEnvKeyTransformer(key string, fn func(string) interface{})
+	// The following helpers allow a type to be enforce when parsing environment variables. Most of them exists to
+	// support historic behavior. Refrain from adding more as it's most likely a sign of poorly design configuration
+	// layout.
+	LoadEnvAsStringSlice(key string, fx func(string) []string)
```

Comment:
The names `LoadEnv...` don't quite work for me since they use callbacks. "Load" implies something happens right now, rather than happening later. How about calling it `UseEnvLoaderAsStringSlice` or something that communicates that it is setting up a function that gets used later.

Reply:
what about `ParseEnvAs` ?

Reply:
Sounds good to me.